clean structure, add signature scanning

ero-qt opened the pull request:
sig scanning should remove all possibility of the script breaking with updates

ero-qt committed:
clean structure, add signature scanning

sig scanning should remove all possibility of the script breaking with updates

neesi committed:
Update LOVE_livesplit.asl

neesi commented:
Thanks for the script, very helpful. I wasn't even aware of signature scanning.

It looks like the framecounter pointer path (0x610, 0x354, 0x120) stops working every time the data.win file is updated unless you're lucky. In the last 10 versions, data.win has changed 6 times. Room pointer sig works through all 10.

I modified your script to obtain the framecounter address using a hook instead. The framecounter signature has changed only once in the last 10 versions (after a major update) and levelactive sig has not changed at all.

I won't merge this yet in case you want to take a look and use a better method / fix things.

ero-qt commented on LOVE/LOVE_livesplit.asl:
```diff
@@ -2,49 +2,214 @@ state("Love") {}
 	vars.Dbg = (Action<dynamic>) ((output) => print("[LOVE ASL] " + output));
 
-	settings.Add("IL_Splits_LOVE", true, "Enable automatic splits for IL mode.");
+	settings.Add("", false);
```
Stop doing this. The settings are not for this. If you need to tell the user some info, put it in a `README.md` on this repo, then link to that in the `LiveSplit.AutoSplitters.xml`.

ero-qt commented on LOVE/LOVE_livesplit.asl:
```diff
@@ -2,49 +2,214 @@ state("Love") {}
+	vars.levelactiveCodecaveAddr = IntPtr.Zero;
+	vars.frameCodecaveAddr = IntPtr.Zero;
+
+	game.Resume(); // why not
```
Why?

ero-qt commented on LOVE/LOVE_livesplit.asl:
```diff
@@ -2,49 +2,214 @@ state("Love") {}
+
+	game.Resume(); // why not
+
+	Thread.Sleep(1000); // hacky way to "fix" occasional update debug errors when you start the game. the errors seem harmless.
```
Sleeping UI thread. No-go.

ero-qt commented on LOVE/LOVE_livesplit.asl:
```diff
@@ -2,49 +2,214 @@ state("Love") {}
+				{
+					game.Resume();
+				}
+
```
Literally what is all of this? Since when do you ever need to write bytes?

ero-qt commented on LOVE/LOVE_livesplit.asl:
```diff
@@ -54,35 +219,103 @@ update
+		{
+			game.Resume();
+		}
+	}
```
Same here, what is this for? What reason could this *possibly* have?

ero-qt commented on LOVE/LOVE_livesplit.asl:
```diff
@@ -92,10 +325,61 @@ isLoading
 exit
 {
-	vars.CancelSource.Cancel();
+	if (vars.ScanThread.IsAlive) vars.CancelSource.Cancel();
```
Unnecessary check.

ero-qt commented on LOVE/LOVE_livesplit.asl:
```diff
@@ -92,10 +325,61 @@ isLoading
+		game.Resume();
+	}
+
+	game.Resume(); // why not
```
Why?

ero-qt commented:
Do not understand the changes. Lots of inconsistencies in casing for the variable names.

neesi committed:
Update LOVE_livesplit.asl

You may have missed my comment here: https://github.com/neesi/autosplitters/pull/1
I should've added that to the previous commit description instead.

It looks like the framecounter pointer path changes every time the data.win file is updated unless you happen to be lucky, and it has been updated 6 times in the last 10 game versions. So instead of using a pointer path, I used aob injection to grab the addresses. This way of doing it would've broke the splitter only once in the last 10 versions (after a major update). I'm sure there is a better / simpler way and I'm all for it.

What I did is more or less the same as this: https://youtu.be/kpSRUJfaT1o?t=726

ero-qt commented:
It's just commonly seen as really bad practice to inject code. I suppose if it works and is completely consistent, then that's fine, but how necessary is this? When was the game last updated? Will it ever be again? Is the game run on any version that isn't the current? Those are just the things you need to weigh to see if this is necessary.

If it is, then I'm fine with the changes and they can be merged.

neesi committed:
Update LOVE_livesplit.asl

Works with at least 10 latest LOVE versions. Didn't test older ones. The signatures work with the dev's other games as well.

neesi committed:
Update LOVE_livesplit.asl

neesi committed:
Update LOVE_livesplit.asl

neesi committed:
Update LOVE_livesplit.asl

neesi committed:
Update LOVE_livesplit.asl

neesi committed:
Update LOVE_livesplit.asl

ero-qt commented:
Could merge at this point, no?
